File: katas/8 kyu/return-negative/solution.py
import codewars_test as test
# -------------------------------- my solution ------------------------------- #
def make_negative( number ):
    return number * -1 if number > 0 else number
# ------------------------------ clever solution ----------------------------- #
def make_negative( number ):
    return -abs(number)
# ----------------------------------- test ----------------------------------- #
@test.describe("Fixed Tests")
def basic_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        for n, expected in ((42,-42), (-9,-9), (0,0)):
            actual = make_negative(n)
            message = f"For n = {n}, expected {expected} but got {actual}"
            test.assert_equals(actual, expected, message)

File: katas/8 kyu/return-negative/solution.php
<!-- return-negative -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function makeNegative(float $num) : float {
  return $num < 0 ? $num : -$num;
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function makeNegative(float $num) : float {
  return -abs($num);
}
/* ---------------------------------- TEST ---------------------------------- */
class MakeNegativeTestCases extends TestCase
{
    // test makeNegative()
    public function testMakeNegative() {
      $this->assertEquals(makeNegative(7), -7);
      $this->assertEquals(makeNegative(-20), -20);
      $this->assertEquals(makeNegative(0), 0);
      $this->assertEquals(makeNegative(0.12), -0.12);
    }
}